Framework/Kernel/src/Property.cpp

```diff
@@ -325,7 +325,10 @@ std::string getUnmangledTypeName(const std::type_info &type) {
   if (mitr != typestrings.end()) {
     return mitr->second;
   }
-
+  string type_name = type.name();
+  if (type_name.find("Mantid") != std::string::npos && type_name.find("EnumeratedString") != std::string::npos) {
```

Comment:
A comment here explaining *why* this is being done will be really helpful in the future.